src/main/java/games/strategy/engine/data/TripleaHandler.java

```diff
@@ -0,0 +1,475 @@
+          break;
+        case "diceSides":
+          handleDiceSides(attributes);
+          break;
+        case XML_NODE_NAME_TERRITORY:
+          handleTerritory(attributes);
+          break;
+        case XML_NODE_NAME_CONNECTION:
+          handleConnection(attributes);
+          break;
+        case XML_NODE_NAME_RESOURCE:
+          handleResource(attributes);
+          break;
+        case XML_NODE_NAME_PLAYER:
+          handlePlayer(attributes);
+          break;
+        case XML_NODE_NAME_ALLIANCE:
+          handleAlliance(attributes);
+          break;
+        case XML_NODE_NAME_UNIT:
+          handleUnit(attributes);
+          break;
+        case XML_NODE_NAME_DELEGATE:
+          handleDelegate(attributes);
+          break;
+        case XML_NODE_NAME_STEP:
+          handleStep(attributes);
+          break;
+        case "playerProduction":
+          handlePlayerProduction(attributes);
+          break;
+        case XML_NODE_NAME_COST:
+          handleCost(attributes);
+          break;
+        case XML_NODE_NAME_RESULT:
+          handleResult(attributes);
+          break;
+        case XML_NODE_NAME_FRONTIER_RULES:
+          handleFrontierRules(attributes);
+          break;
+        case XML_NODE_NAME_OPTION:
+          handleOption(attributes);
+          break;
+        case XML_NODE_NAME_TERRITORY_OWNER:
+          handleTerritoryOwner(attributes);
+          break;
+        case XML_NODE_NAME_UNIT_PLACEMENT:
+          handleUnitPlacement(attributes);
+          break;
+        case "resourceGiven":
+          handleResourceGiven(attributes);
+          break;
+        case XML_NODE_NAME_PROPERTY:
+          handleProperty(attributes);
+          break;
+        case XML_NODE_NAME_NUMBER:
+          handleNumber();
+          break;
+        case XML_NODE_NAME_BOOLEAN:
+          handleBoolean();
+          break;
+        case "tech":
+          handleTech(attributes);
+          break;
+        case XML_NODE_NAME_PRODUCTION_FRONTIER:
+          handleProductionFrontier(attributes);
+          break;
+        case XML_NODE_NAME_ATTACHMENT:
+          handleAttachment(attributes);
+          break;
+        case XML_NODE_NAME_VALUE:
+        case XML_NODE_NAME_SEQUENCE:
+        case XML_NODE_NAME_RESOURCE_LIST:
+        case XML_NODE_NAME_UNIT_LIST:
+        case "relationshipTypes":
+        case "territoryEffectList":
+        case "technology":
+        case XML_NODE_NAME_OWNER_INITIALIZE:
+        case XML_NODE_NAME_UNIT_INITIALIZE:
+        case "resouceInitialize":
+        case "technologies":
+        case "playerTech":
+        case "category":
+        default:
+          break;
+      }
+    } catch (GameParseException e) {
+      throw new SAXException(e);
+    }
+  }
+
+  private void checkParentNode(String qname) throws GameParseException {
+    if (!stack.isEmpty()) {
+      String actualName = stack.peek().getName();
+      Collection<String> parents = requiredParents.get(qname);
```

Comment:
If the xml is well-formed and semantically correct nothing would change at all.
Otherwhise we would probably run into a lot of "missing attribute for name" exceptions which would make it harder to find the real issue.
So this method is basically a fail early concept